Source/Spatializer/CrossPlatform/SpatializerPlugin.cpp

```diff
@@ -1,402 +1,254 @@
-        }
-
-        auto S = state->spatializerdata->sourcematrix;
-        auto L = state->spatializerdata->listenermatrix;
-
-        // If we previously released the source, get one back
-        if (data->EffectHrtfInfo == nullptr)
-        {
-            data->EffectHrtfInfo.reset(HrtfWrapper::GetHrtfSource());
-
-            // If EffectHrtfInfo is still null, that means we're not able to get HRTF resources.
-            // Mute this source to prevent unexpectedly loud sounds
-            if (data->EffectHrtfInfo == nullptr)
-            {
-                memset(outbuffer, 0, length * outChannels * sizeof(float));
-                return UNITY_AUDIODSP_OK;
-            }
-        }
-
-        // There's no acoustics support yet, update params using a through-the-wall method
-        UpdateAcousticParams(data, state->spatializerdata->spread, ListenerToSourceDirection(S, L));
-
-        // Sometimes, the previous allocation can fail and produce a SourceBuffer with a null array
-        // Make sure we have a buffer to use before proceeding
-        if (data->EffectHrtfInfo->GetBuffer())
-        {
-            PrepareAudioData(state, inbuffer, outbuffer, length, outChannels);
-        }
-
-        return UNITY_AUDIODSP_OK;
-    }
-} // namespace Spatializer
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Please note that this will only work on Unity 5.2 or higher.
+
+#include "AudioPluginUtil.h"
+#include "HelperMacros.h"
+#include "HrtfWrapper.h"
+#include "VectorMath.h"
+#include "mathutility.h"
+
+#include <math.h>
+#include <algorithm>
+#include <memory>
+#include <cstring>
+
+namespace Spatializer
+{
+    enum EffectParams
+    {
+        Count
+    };
+
+    struct EffectData
+    {
+        std::unique_ptr<HrtfWrapper::SourceInfo> EffectHrtfInfo;
+        float SourceDistance;
+        float DryDistanceAttenuation;
+    };
+
+    int InternalRegisterEffectDefinition(UnityAudioEffectDefinition& definition)
+    {
+        definition.flags |= UnityAudioEffectDefinitionFlags_IsSpatializer;
+        return 0;
+    }
+
+    static UNITY_AUDIODSP_RESULT UNITY_AUDIODSP_CALLBACK DistanceAttenuationCallback(
+        UnityAudioEffectState* state, float distanceIn, float attenuationIn, float* attenuationOut)
+    {
+        // Tell Unity not to apply any attenuation, since we will render the specified attenuation on the dry path
+        *attenuationOut = 1.0f;
+        if (attenuationIn < c_MinAudibleGain)
+        {
+            // If source is quiet, tell Unity to mute it
+            *attenuationOut = 0.0f;
+        }
+
+        // Save off this data so we can use it later
+        auto data = state->GetEffectData<EffectData>();
+        data->SourceDistance = distanceIn;
+        data->DryDistanceAttenuation = attenuationIn;
+        return UNITY_AUDIODSP_OK;
+    }
+
+    UNITY_AUDIODSP_RESULT UNITY_AUDIODSP_CALLBACK CreateCallback(UnityAudioEffectState* state)
```

Comment:
I think you're missing the call to InternalRegisterEffectDefinition here

Reply:
Ah yeah, I put it bak although what does it do? Cause I didn't see any difference in behavior.

Reply:
Marks this plugin as a spatializer. If you're not hearing a difference, it makes me wonder if some other stuff is preventing the plugin from functioning properly, or if Unity is somehow finding another way to treat this plugin as a spatializer.